lib/common/errors.ts

```diff
@@ -123,14 +124,33 @@ export class Errors implements IErrors {
 
 	public printCallStack: boolean = false;
 
-	public fail(optsOrFormatStr: any, ...args: any[]): never {
-		const argsArray = args || [];
+	public fail(optsOrFormatStr: string | IFailOptions, ...args: any[]): never {
+		const opts = this.getFailOptions(optsOrFormatStr);
+		return this.failWithOptions(opts, false, args);
```

Comment:
I think you need to pass `...args` here instead of just `args`. With the current way, you pass the whole args array as a single element. After that, in the `failWithOptions` you are using the rest operator, which will place the passed argument in an array, so `args` in `failWithOptions` method will be array with a single element, which is an array.
More information here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters#Description